cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -476,75 +476,74 @@ func TestDrain(t *testing.T) {
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&rc},
 			expectFatal: true,
 			expectPods:  []*apiv1.Pod{},
 		},
 		{
 			description: "empty PDB with RC-managed pod",
 			pods:        []*apiv1.Pod{rcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{emptyPDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&rc},
 			expectFatal: false,
 			expectPods:  []*apiv1.Pod{rcPod},
 		},
 		{
 			description: "kube-system PDB with matching kube-system pod",
 			pods:        []*apiv1.Pod{kubeSystemRcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{kubeSystemPDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&kubeSystemRc},
 			expectFatal: false,
 			expectPods:  []*apiv1.Pod{kubeSystemRcPod},
 		},
 		{
 			description: "kube-system PDB with non-matching kube-system pod",
 			pods:        []*apiv1.Pod{kubeSystemRcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{kubeSystemFakePDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&kubeSystemRc},
 			expectFatal: true,
 			expectPods:  []*apiv1.Pod{},
 		},
 		{
 			description: "kube-system PDB with default namespace pod",
 			pods:        []*apiv1.Pod{rcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{kubeSystemPDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&rc},
 			expectFatal: false,
 			expectPods:  []*apiv1.Pod{rcPod},
 		},
 		{
 			description: "default namespace PDB with matching labels kube-system pod",
 			pods:        []*apiv1.Pod{kubeSystemRcPod},
 			pdbs:        []*policyv1.PodDisruptionBudget{defaultNamespacePDB},
-			rcs:         []apiv1.ReplicationController{rc},
+			rcs:         []*apiv1.ReplicationController{&kubeSystemRc},
 			expectFatal: true,
 			expectPods:  []*apiv1.Pod{},
 		},
 	}
 
 	for _, test := range tests {
-		fakeClient := &fake.Clientset{}
-		register := func(resource string, obj runtime.Object, meta metav1.ObjectMeta) {
-			fakeClient.Fake.AddReactor("get", resource, func(action core.Action) (bool, runtime.Object, error) {
-				getAction := action.(core.GetAction)
-				if getAction.GetName() == meta.GetName() && getAction.GetNamespace() == meta.GetNamespace() {
-					return true, obj, nil
-				}
-				return false, nil, fmt.Errorf("Not found")
-			})
-		}
+		var err error
+		var rcLister v1lister.ReplicationControllerLister
 		if len(test.rcs) > 0 {
-			register("replicationcontrollers", &test.rcs[0], test.rcs[0].ObjectMeta)
+			rcLister, err = kube_util.NewTestReplicationControllerLister(test.rcs)
+			assert.NoError(t, err)
 		}
-		register("daemonsets", &ds, ds.ObjectMeta)
-		register("jobs", &job, job.ObjectMeta)
-		// register("statefulsets", &statefulset, statefulset.ObjectMeta)
-
+		var rsLister v1appslister.ReplicaSetLister
 		if len(test.replicaSets) > 0 {
```

Comment:
can we get rid of this checks? Are they needed?

Reply:
I just left this part as it was, but it's a bit of a mess. RC and RS are passed as test param, but other controllers are just always set. I'd rather avoid cleaning up all those issues from tests as it will bloat this PR. I can open an issue to clean it up later, though https://github.com/kubernetes/autoscaler/issues/215 illustrates the problem with that (I only re-enabled the test it mentions now as I was making changes in this area).

Reply:
I am not opting for fixing everything. But if we can clean something up, as part adding feature, I am for it. I know you prefer the PRs to address only one thing but the issue is that with that approach the cleanup PRs never happen. On the other hand if can clean just a bit, as part of feature-PR, we are a bit more in the better world.